src/server/telemetry.js

```diff
@@ -0,0 +1,168 @@
+var http = require('http');
+var https = require('https');
+var _ = require('../utility');
+var urlHelpers = require('./telemetry/urlHelpers');
+
+var defaults = {
+  network: true,
+  networkResponseHeaders: false,
+  networkRequestHeaders: false,
+  log: true
+};
+
+function Instrumenter(options, telemeter, rollbar) {
+  this.options = options;
+  var autoInstrument = options.autoInstrument;
+  if (options.enabled === false || autoInstrument === false) {
+    this.autoInstrument = {};
+  } else {
+    if (!_.isType(autoInstrument, 'object')) {
+      autoInstrument = defaults;
+    }
+    this.autoInstrument = _.merge(defaults, autoInstrument);
+  }
+  this.telemeter = telemeter;
+  this.rollbar = rollbar;
+  this.diagnostic = rollbar.client.notifier.diagnostic;
+  this.replacements = {
+    network: [],
+    log: []
+  };
+}
+
+Instrumenter.prototype.configure = function(options) {
+  this.options = _.merge(this.options, options);
+  var autoInstrument = options.autoInstrument;
+  var oldSettings = _.merge(this.autoInstrument);
+  if (options.enabled === false || autoInstrument === false) {
+    this.autoInstrument = {};
+  } else {
+    if (!_.isType(autoInstrument, 'object')) {
+      autoInstrument = defaults;
+    }
+    this.autoInstrument = _.merge(defaults, autoInstrument);
+  }
```

Comment:
Ditto with my comment above about clarifying the deep nesting here.

Reply:
:+1: